### PR TITLE
Fix issue #168 optionals without stats

### PR DESCRIPTION
**Pull request by Teun**
### Fixes

Issue #168 and maybe also issue #164

### Description

- [x] I have included unit tests validating this fix.
- [x] I have updated markdown documentation where required.
- [x] I understand that successful approval of my pull request requires reproducible tests as per [Contribution Guideline](https://github.com/aloneguid/parquet-dotnet/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Teun** [comment]
Added a unit test proving that #164 is also fixed with this PR

File: src/Parquet/File/DataColumnReader.cs [hunk 1]
```diff
@@ -204,16 +205,19 @@ private void ReadDataPage(Thrift.PageHeader ph, ColumnRawData cd, long maxValues
                      if (cd.definitions == null)
                         cd.definitions = new int[cd.maxCount];
 
+                     int offsetBeforeReading = cd.definitionsOffset;
                      cd.definitionsOffset += ReadLevels(reader, _maxDefinitionLevel, cd.definitions, cd.definitionsOffset, ph.Data_page_header.Num_values);
+                     valueCount = cd.definitions.Skip(offsetBeforeReading).Take(cd.definitionsOffset - offsetBeforeReading)
```
Comment (kzrodlowski) [inline src/Parquet/File/DataColumnReader.cs:210]: Could we avoid the value count calculations if data page header statistics are provided? We don't really need this value when stats are available.

Comment (Teun) [reply src/Parquet/File/DataColumnReader.cs]: Good point. Changed it.

**Review by kzrodlowski — DISMISSED** [review]

**Comment by Teun** [comment]
Mystery to me why that Á in the literal string caused issues. Looks fine in my branch, looks fine after the PR merge, but indeed, when I check out the code on a mac, it garbles the string. Have replaced it by a \u reference and now it works fine on the mac.

**Review by kzrodlowski — APPROVED** [review]

**Comment by Teun** [comment]
@aloneguid could you please make he CI actions run again? The pushed changes should have fixed the MacOS problem.

**Review by aloneguid — APPROVED** [review]

**Comment by aloneguid** [comment]
@Teun not sure how to trigger CI, it should just kick off automatically.

**Comment by kzrodlowski** [comment]
@Teun is there any chance we could proceed with these changes?

**Comment by Teun** [comment]
I'd love to, but I'm not sure about the process. I provided a PR, the Test action found a problem. Then I pushed a fix for that issue, but the action never ran again. Maybe someone with the required privileges can run it manually?

**Comment by kzrodlowski** [comment]
> I'd love to, but I'm not sure about the process. I provided a PR, the Test action found a problem. Then I pushed a fix for that issue, but the action never ran again. Maybe someone with the required privileges can run it manually?

@aloneguid could I ask you for help with driving this PR to completion? If there's no easy way to trigger it, should the PR be re-submitted?

**Comment by shamimashik** [comment]
Does this fix exist in the latest version (4.23.x)? I'm trying to update from 3.10.x to the latest and seeing failure around this. Can someone confirm?

**Pull request merged**